src/components/dropdown/DropdownPlan.jsx

```diff
@@ -68,14 +77,57 @@ const DropdownPlan = ({ title, date, location, description, author }) => {
         }
     };
 
+    const onAddToGoogleCalendar = () => {
+        const [day, month, year] = date.split("/");
+        const dateObject = new Date(`${year}-${month}-${day}`);
+
+        const encodedUrl = encodeURI(
+            [
+                "https://www.google.com/calendar/render",
+                "?action=TEMPLATE",
+                `&text=${title || ""}`,
+                `&dates=${dateObject || ""}`,
+                // TODO: Calculate (duration + dateObject) to get the endDate
+                //  `/${  endDate || ''}`,
+                `&details=${description || ""}`,
+                `&location=${location || ""}`,
+                "&sprop=&sprop=name:",
+            ].join(""),
+        );
+        return encodedUrl;
+    };
+
+    const onAddToAppleCalendar = () => {
+        const [day, month, year] = date.split("/");
+        const dateObject = new Date(`${year}-${month}-${day}`);
```

Comment:
the code is the same, only changes the url. I think can be abstracted to only one unique function

Reply:
Solved